components/switch/switch-mixin.js

```diff
@@ -0,0 +1,181 @@
+			}
+
+			.d2l-switch-toggle {
+				position: relative;
+				transition: transform 150ms linear;
+			}
+			.d2l-switch-toggle > div {
+				background-color: #ffffff;
+				border: 1px solid var(--d2l-color-ferrite);
+				border-radius: 0.6rem;
+				box-sizing: border-box;
+				display: inline-block;
+				height: 1.2rem;
+				left: -0.2rem;
+				position: absolute;
+				top: -0.2rem;
+				width: 1.2rem;
+			}
+			:host([dir="rtl"]) .d2l-switch-toggle > div {
+				left: auto;
+				right: -0.2rem;
+			}
+			:host([on]) .d2l-switch-toggle {
+				transform: translateX(1.2rem);
+			}
+			:host([dir="rtl"][on]) .d2l-switch-toggle {
+				transform: translateX(-1.2rem);
+			}
+			:host([on]) .d2l-switch-toggle > div {
+				border-color: var(--d2l-color-celestine);
+			}
+
+			d2l-icon, d2l-icon-custom {
+				height: 0.8rem;
+				width: 0.8rem;
+			}
+			.d2l-switch-icon-on, .d2l-switch-icon-off {
+				display: inline-block;
+			}
+			.d2l-switch-icon-on {
+				margin-right: 0.4rem;
+			}
+			:host([dir="rtl"]) .d2l-switch-icon-on {
+				margin-left: 0.4rem;
+				margin-right: 0;
+			}
+			:host([on]) .d2l-switch-icon-on > d2l-icon,
+			:host([on]) .d2l-switch-icon-on > d2l-icon-custom {
+				color: var(--d2l-color-celestine);
+			}
+
+			.d2l-switch-label {
+				font-size: 0.8rem;
+				font-weight: 400;
+			}
+
+			@media (prefers-reduced-motion: reduce) {
+				.d2l-switch-toggle {
+					transition: none;
+				}
+			}
+		`;
+	}
+
+	constructor() {
+		super();
+		this.disabled = false;
+		this.labelHidden = false;
+		this.on = false;
+		this._labelId = getUniqueId();
+	}
+
+	render() {
+		return html`
+			<div
+				aria-checked="${this.on ? 'true' : 'false'}"
+				aria-label="${ifDefined(this.labelHidden ? this.label : undefined)}"
+				aria-labelledby="${ifDefined(!this.labelHidden ? this._labelId : undefined)}"
+				class="d2l-switch-container"
+				@click="${this._handleClick}"
+				@keydown="${this._handleKeyDown}"
+				@keyup="${this._handleKeyUp}"
+				role="switch"
+				tabindex="0">
+				<div class="d2l-switch-inner">
+					<div class="d2l-switch-toggle"><div></div></div>
+					<div class="d2l-switch-icon-on">${this.onIcon}</div>
+					<div class="d2l-switch-icon-off">${this.offIcon}</div>
+				</div>
+			</div>
+			${!this.labelHidden ? html`<span id="${this._labelId}" class="d2l-switch-label">${this.label}</span>` : ''}
+		`;
+	}
+
+	updated(changedProperties) {
+		super.updated(changedProperties);
+		if (!changedProperties.has('on') || changedProperties.get('on') === undefined) return;
+		this.dispatchEvent(new CustomEvent('change', { bubbles: true }));
+	}
+
+	_handleClick() {
+		this._toggleState();
+	}
+
+	_handleKeyDown(e) {
+		// space pressed... prevent default browser scroll
+		if (e.keyCode === 32) e.preventDefault();
+	}
+
+	_handleKeyUp(e) {
+		// space pressed... toggle state
+		if (e.keyCode === 32) this._toggleState();
+	}
+
+	_toggleState() {
+		if (this.disabled) return;
+		this.on = !this.on;
```

Comment:
Mostly I'm curious: what should happen if folks manually change the "on" state by setting the property or attribute themselves when the component is disabled? I just tried this with checkboxes and it goes ahead with the state change, which I think is correct since if you're mucking around in JavaScript you probably know what you're doing. And that matches your current behavior, but might be worth ensuring there's a test for it.